### PR TITLE
Editor component

### LINKED ISSUE #4: Front-End : Create a web-page with a basic editor

### PR DESCRIPTION
**Pull request by maurelio1234**
Fixes #4 

image

### CODE REVIEW

**Comment by maurelio1234** [comment]
This is a first take on it, since the ticket is a little bit low on details, let's start with that.

Please tell me if you have some other things in mind in your `definition of done`.

File: webapp/src/App.tsx [hunk 1]
```diff
@@ -1,20 +1,68 @@
...
       <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h1 className="App-title">Welcome to <FormattedMessage id="app.title" /></h1>
-        </header>
-        <p className="App-intro">
-          To get started, edit <code>src/App.tsx</code> and save to reload.
-        </p>
+        <div id="topSection" style={{ height: '44px' }}>
+          <div id="selector" style={{ top: '11px', position: 'absolute', right: '0px'}} >
```
Comment (maurelio1234) [inline webapp/src/App.tsx]: externalise style

File: webapp/src/App.tsx [hunk 2]
```diff
@@ -1,20 +1,68 @@
...
+                <option value="csharp">C#</option>
+                <option value="java">Java</option>
+                <option value="python">Python</option>
+                <option value="javascript">JavaScript</option>
+                <option value="typescript">TypeScript</option>
+              </select>
+            </label>
+          </div>
+        </div>
+        <AceEditor
```
Comment (maurelio1234) [inline webapp/src/App.tsx:47]: keep source code in state too

File: webapp/jest.config.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
```
Comment (maurelio1234) [inline webapp/jest.config.js]: consider removing jest related dependencies

**Review by nicolaserny — APPROVED** [review]
Optional: you can create components (parameterHeader and SourceCodeEditor for example) instead of using a single App component.

**Comment by nicolaserny** [comment]
@Julien-Molina Are you all right with this definition of done?

**Comment by maurelio1234** [comment]
@nicolaserny 

I have mixed feelings about creating different components... it seems overkill to me...
What about encapsulating the header + editor in a single component? 
(it is still overkill, since App wil be composed of a single sub component...)

**Comment by nicolaserny** [comment]
@maurelio1234  That's why it is optional so far. Creating small components improves rendering performance. But we will do it (if necessary) later on.

**Pull request merged**